batch-index-files/main.py

```diff
@@ -8,55 +8,81 @@
 from boto3 import Session
 
 import bioindex.lib.s3 as s3
+import requests
+import threading
 
 
 @click.command()
+@click.option('--index', '-i', type=str)
 @click.option('--bucket', '-b', type=str)
 @click.option('--path', '-p', type=str)
 @click.option('--delete', '-d', type=bool, default=False)
-def main(bucket, path, delete):
+def main(index, bucket, path, delete):
+    print("starting")
     keys = get_access_keys()
     # transfer the secrets to environment variables (keep sensitive info out of source control)
     os.environ['AWS_ACCESS_KEY_ID'] = keys['access_key_id']
     os.environ['AWS_SECRET_ACCESS_KEY'] = keys['secret_access_key']
     s3_objects = list(s3.list_objects(bucket, path, only='*.json'))
     boto_s3 = boto3.client('s3')
     print(f"will compress {len(s3_objects)} files")
+    files_to_retry = process_files_concurrently(boto_s3, bucket, delete, [file['Key'] for file in s3_objects])
+    if len(files_to_retry) > 0:
```

Comment:
I've noticed that bgzip occasionally fails, but retries have thus far always cleaned things up.  This code will take one pass at bgzipping all files, capture those that failed and then take one more run through at those.